File: Sorting/BubbleSort.cpp
#include <iostream>
#include <vector>
#include<algorithm>

void bubbleSort(std::vector<int>& arr)
{
    int n = arr.size();
    for (int i = 0; i < n - 1; ++i)
    {
        for (int j = 0; j < n - i - 1; ++j)
        {
            if (arr[j] > arr[j + 1])
            {
                std::swap(arr[j], arr[j + 1]);
            }
        }
    }
}

void printArray(const std::vector<int>& arr)
{
    for (int x : arr)
    {
        std::cout << x << " ";
    }
    std::cout << std::endl;
}

int main()
{
    std::vector<int> data = {64, 34, 25, 12, 22, 11, 90};

    std::cout << "Original array: ";
    printArray(data);

    bubbleSort(data);

    std::cout << "Sorted array: ";
    printArray(data);

    return 0;
}

File: Sorting/BubbleSort.py
def bubbleSort(arr):
    n = len(arr)

    for i in range(n):
        swapped = False  

        for j in range(0, n - i - 1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True

        if not swapped:
            break
    return arr

myList = [64, 34, 45, 92, 17, 1, 29, 55]
SortedList = bubbleSort(myList)
print(SortedList)
